### ISSUE TITLE
fish shell autocomplete error for zfs

### ISSUE THREAD

**Issue by devvythelopper** [NixOS/nixpkgs#247290]
### Describe the bug

when using zfs autocomplete in the fish shell, one tends to get an error: `_: cannot overwrite read-only variable` which totally messes up the shell's contents. There is a fish-shell issue concerning this, in which a fix is described (https://github.com/fish-shell/fish-shell/issues/9705), which consists of simply overwriting the buggy `/usr/share/fish/functions/__fish_is_zfs_feature_enabled.fish`. 

### Steps To Reproduce
1. use fish
2. use zfs command triggering autocomplete

### Expected behavior
No error.

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

**Comment by Cu3PO42** [comment]
I can confirm this issue. However, this is a bug in upstream, which they have fixed but not yet cut a release for. It should be fixed as soon as they release either 3.6.2 or 3.7.0 and nixpkgs updates to that version. In the meantime, I have chosen to simply apply the patch manually in my config. A possible way to do that is as follows:
```
  nixpkgs.overlays = [
    (final: prev: {
      fish = prev.fish.overrideAttrs (o: {
      patches = (o.patches or [ ]) ++ [
        (pkgs.fetchpatch {
          name = "fix-zfs-completion.path";
          url = "https://github.com/fish-shell/fish-shell/commit/85504ca694ae099f023ae0febb363238d9c64e8d.patch";
          sha256 = "sha256-lA0M7E/Z0NjuvppC7GZA5rWdL7c+5l+3SF5yUe7nEz8=";
        })
      ];
    });
    })
  ];
```

**Comment by pbek** [comment]
3.6.4 didn't fix it, I still have the same error.

**Comment by iynaix** [comment]
Should be fixed in 3.7.0
